tests/analyze.py

```diff
@@ -0,0 +1,43 @@
+from collections import defaultdict
```

Comment:
what's this?

Reply:
Yes, sorry, I was blindly pushing for CI, forgot to untag your review